managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedCursor.java

```diff
@@ -86,14 +86,20 @@ enum IndividualDeletedEntries {
      */
     Map<String, String> getCursorProperties();
 
-     /**
-      * Updates the properties.
-      * @param cursorProperties
-      * @return a handle to the result of the operation
-      */
-     default CompletableFuture<Void> setCursorProperties(Map<String, String> cursorProperties) {
-         return CompletableFuture.completedFuture(null);
-     }
+    /**
+     * Add a property associated with the cursor.
+     */
+    void putCursorProperty(String key, String value);
+
+    /**
+     * Set all properties associated with the cursor, internal properties are still preserved.
+     */
+    void setAllCursorProperties(Map<String, String> properties);
+
+    /**
+     * Remove a property associated with the cursor.
+     */
+    void removeCursorProperty(String key);
 
```

Comment:
Instead of modifying the interface, it would be better to fix the issue in the existing method, `CompletableFuture<Void> setCursorProperties(Map<String, String> cursorProperties)`.

Adding a new "blocking API" instead of using an asynchronous method will cause issues.

Reply:
> Instead of modifying the interface, it would be better to fix the issue in the existing method, `CompletableFuture<Void> setCursorProperties(Map<String, String> cursorProperties)`.
> 
> Adding a new "blocking API" instead of using an asynchronous method will cause issues.

If keep using an asynchronous method will not ensure data consistency when the concurrent modification.

Reply:
> What's wrong with using the synchronous method?

The main reason to avoid blocking methods in Pulsar is to prevent blocking Netty IO threads. Netty uses a small number of IO threads to execute and using blocking on these threads could cause the execution of work assigned to Netty event loops to not proceed as expected (other connections might hang / timeout / get delayed etc.). 
There are also other reasons to avoid synchronous methods such as "[PIP-149 Making the REST Admin API fully async](https://github.com/apache/pulsar/issues/14365)". (cc @Technoboy- and @mattisonchao)

Reply:
> If keep using an asynchronous method will not ensure data consistency when the concurrent modification.

It should be possible to resolve this issue without changing the API.

Reply:
> > What's wrong with using the synchronous method?
> 
> The main reason to avoid blocking methods in Pulsar is to prevent blocking Netty IO threads. Netty uses a small number of IO threads to execute and using blocking on these threads could cause the execution of work assigned to Netty event loops to not proceed as expected (other unrelated connections might hang / timeout / get delayed etc.). There are also other reasons to avoid synchronous methods such as "[PIP-149 Making the REST Admin API fully async](https://github.com/apache/pulsar/issues/14365)". (cc @Technoboy- and @mattisonchao)

I see, I think can keep `CompletableFuture<Void> setCursorProperties(Map<String, String> cursorProperties)` asynchronous method, but need first update variable before update store.
And add `CompletableFuture<Void> putCursorProperty(String key, String value)` and `CompletableFuture<Void> setCursorProperties(Map<String, String> properties)`